wlauto/utils/statedetect.py

```diff
@@ -0,0 +1,120 @@
+#    Copyright 2013-2016 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+"""
+State detection functionality for revent workloads. Uses OpenCV to analyse screenshots from the device.
+Requires a 'statedetection' directory in the workload directory that includes the state definition yaml file,
+and the 'templates' folder with PNGs of all templates mentioned in the yaml file.
+
+Requires the following python plugins:
+numpy, pyyaml (yaml), imutils and opencv (cv2)
```

Comment:
Please add the new dependencies to setup.py.

Also, since some of these are quite "heavy" (in particular we do *not* want to depend on numpy by default), these should be added as optional dependencies in their own group. Which means that here, you cannot assume that the imports will succeed. You need to guard against that and report and error only if the user tried to use the optional functionality without those modules installed. 

For example, see `fps` instrument which uses the optional `pandas` dependency:

https://github.com/ARM-software/workload-automation/blob/master/wlauto/instrumentation/fps/__init__.py